fuse/ipns/ipns_unix.go

```diff
@@ -285,7 +285,7 @@ func (s *Node) Attr() fuse.Attr {
 }
 
 // Lookup performs a lookup under this node.
-func (s *Node) Lookup(name string, intr fs.Intr) (fs.Node, fuse.Error) {
+func (s *Node) Lookup(name string, ctx context.Context) (fs.Node, error) {
```

Comment:
@cryptix I don't know where you got that signature from. It clearly doesn't implement http://godoc.org/bazil.org/fuse/fs#NodeStringLookuper or http://godoc.org/bazil.org/fuse/fs#NodeRequestLookuper -- I would expect all those lookups to fail (no lookup function exists), and your code working because of a combination of two things: 1. cached dentries from Create etc, and 2. lack of test coverage.

Do this:

```go
var _ fs.NodeStringLookuper = (*Node)(nil)
```

for every interface you want to make sure you implement.

Maybe once you've fixed it, drop an obvious beacon like a log message or panic in the function, and improve your tests until it triggers?